docs/src/pages/components/data-grid/columns/ValueParserGrid.js

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import { DataGrid } from '@material-ui/data-grid';
+
+const rows = [
+  {
+    id: 1,
+    date: new Date(1979, 0, 1),
+  },
+  {
+    id: 2,
+    date: new Date(1984, 0, 1),
+  },
+  {
+    id: 3,
+    date: new Date(1992, 0, 1),
+  },
+];
+
+export default function ValueParserGrid() {
+  return (
+    <div style={{ height: 300, width: '100%' }}>
+      <DataGrid
+        rows={rows}
+        columns={[
+          {
+            type: 'number',
+            field: 'date',
+            headerName: 'Year',
+            width: 150,
+            valueFormatter: (params) => params.value.getFullYear(),
+            valueParser: (value) => new Date(Number(value), 0, 1),
```

Comment:
Should we make it editable, so we can also test the editability with this demo?

Reply:
If I make it editable, I'll have to use a custom component for the edit cell because the default one is converting the value to a date string (when it's a `Date`) and that doesn't work with `input[type=number]`. I would prefer to leave as it is to keep the code below the demo's toolbar simple.

Reply:
it feels like the valueFormatter should be a valueGetter as you are changing the type of value.